connectors/sources/outlook.py

```diff
@@ -0,0 +1,1003 @@
+
+MAIL_TYPES = [
+    {
+        "folder": "inbox",
+        "constant": INBOX_MAIL_OBJECT,
+    },
+    {
+        "folder": "sent",
+        "constant": SENT_MAIL_OBJECT,
+    },
+    {
+        "folder": "junk",
+        "constant": JUNK_MAIL_OBJECT,
+    },
+    {
+        "folder": "archive",
+        "constant": ARCHIVE_MAIL_OBJECT,
+    },
+]
+
+MAIL_FIELDS = [
+    "sender",
+    "to_recipients",
+    "cc_recipients",
+    "bcc_recipients",
+    "last_modified_time",
+    "subject",
+    "importance",
+    "categories",
+    "body",
+    "has_attachments",
+    "attachments",
+]
+CONTACT_FIELDS = [
+    "email_addresses",
+    "phone_numbers",
+    "last_modified_time",
+    "display_name",
+    "company_name",
+    "birthday",
+]
+TASK_FIELDS = [
+    "last_modified_time",
+    "due_date",
+    "complete_date",
+    "subject",
+    "status",
+    "owner",
+    "start_date",
+    "text_body",
+    "companies",
+    "categories",
+    "importance",
+    "has_attachments",
+    "attachments",
+]
+CALENDAR_FIELDS = [
+    "required_attendees",
+    "type",
+    "recurrence",
+    "last_modified_time",
+    "subject",
+    "start",
+    "end",
+    "location",
+    "organizer",
+    "body",
+    "has_attachments",
+    "attachments",
+]
+
+END_SIGNAL = "FINISHED"
+
+
+def ews_format_to_datetime(source_datetime, timezone):
+    """Change datetime format to user account timezone
+    Args:
+        datetime: Datetime in UTC format
+        timezone: User account timezone
+    Returns:
+        Datetime: Date format as user account timezone
+    """
+    if isinstance(source_datetime, exchangelib.ewsdatetime.EWSDateTime):
+        return (source_datetime.astimezone(pytz.timezone(str(timezone)))).strftime(
+            "%Y-%m-%dT%H:%M:%SZ"
+        )
+    elif isinstance(source_datetime, exchangelib.ewsdatetime.EWSDate) or isinstance(
+        source_datetime, date
+    ):
+        return source_datetime.strftime("%Y-%m-%d")
+    else:
+        return source_datetime
+
+
+class TokenFetchFailed(Exception):
+    """Exception class to notify that connector was unable to fetch authentication token from Graph API"""
+
+    pass
+
+
+class UsersFetchFailed(Exception):
+    """Exception class to notify that connector was unable to fetch users from Active Directory"""
+
+    pass
+
+
+class UnauthorizedException(Exception):
+    """Exception class unauthorized calls"""
+
+    pass
+
+
+class Forbidden(Exception):
+    pass
+
+
+class NotFound(Exception):
+    pass
+
+
+class SSLFailed(Exception):
+    pass
+
+
+class ManageCertificate:
+    cert_file = "outlook_cert.cer"
+
+    def store_certificate(self, certificate):
+        with open(self.cert_file, "w") as file:
+            file.write(certificate)
+
+    def get_certificate_path(self):
+        return os.path.join(os.getcwd(), self.cert_file)
+
+    def remove_certificate_file(self):
+        if os.path.exists(self.cert_file):
+            os.remove(self.cert_file)
```

Comment:
we can use to make this non-blocking
```
from aiofiles.os import remove
await remove(...)
```

Reply:
`aiofiles` module does not have `os` attribute, so need to keep this.